### ISSUE TITLE
UX Audit and Recommendations

### ISSUE THREAD

**Issue by glenn-sorrentino** [EFForg/https-everywhere#16669]
[//]: # (Thank you for reporting an issue to HTTPS Everywhere.)
[//]: # (We welcome input from users on improving this project.)
[//]: #
[//]: # (Please help us by following this issue template.)
[//]: # (You can delete all blank lines and all lines starting)
[//]: # (with the comment marker, such as this one.)
[//]: #
[//]: # (Delete all but one of the following "Type" lines.)
[//]: # (Leave in the line that best describes the issue you are reporting.)
[//]: #
[//]: # (If you are submitting a new ruleset, please check the list at)
[//]: # (https://www.eff.org/https-everywhere/atlas/index.html and the open)
[//]: # (issues and pull requests to make sure it doesn't already exist.)

Type: other

Hi everyone, I'm Glenn. I've contributed UX and visual design to Signal, Haven, and OnionShare in the past, and I put together a [UX audit and design recommendations](https://github.com/EFForg/https-everywhere/files/2442586/HTTPS.Everywhere.UX.pdf) for HTTPS Everywhere. I'd love your thoughts and feedback...

**The Problem - UX Hypothesis**
The functionality within the extension panel uses mixed metaphors for the UI and inconsistent hierarchical groupings of functionality which can lead to confusion and frustration for users and create barriers to interaction or adoption.

**The Problem - Visual Hypothesis**
With the evolving UX of the competitive landscape, the design language and UI of HTTPS Everywhere has become dated.

**The Opportunity**
By creating hierarchical groupings of functionality and by using appropriate controls in the UI, we can create a highly usable and beautiful user experience.

[Download updated UX audit](https://github.com/EFForg/https-everywhere/files/2442586/HTTPS.Everywhere.UX.pdf)

![https everywhere ux 19](https://user-images.githubusercontent.com/28545431/46419670-3e4e6100-c6e3-11e8-8350-09ad656bbe1b.jpg)

![https everywhere ux 1](https://user-images.githubusercontent.com/28545431/46419237-51acfc80-c6e2-11e8-9838-3fff7a5fe2d5.jpg)

![https everywhere ux 2](https://user-images.githubusercontent.com/28545431/46419238-51acfc80-c6e2-11e8-9def-14b71c7f254c.jpg)

![https everywhere ux 3](https://user-images.githubusercontent.com/28545431/46419239-52459300-c6e2-11e8-832d-f9aab3936ff7.jpg)

![https everywhere ux 4](https://user-images.githubusercontent.com/28545431/46419240-52de2980-c6e2-11e8-9df2-253749e27cd4.jpg)

![https everywhere ux 5](https://user-images.githubusercontent.com/28545431/46419242-52de2980-c6e2-11e8-8cfc-d820333bd309.jpg)

![https everywhere ux 6](https://user-images.githubusercontent.com/28545431/46419243-52de2980-c6e2-11e8-93de-9deb51487dfc.jpg)

![https everywhere ux 7](https://user-images.githubusercontent.com/28545431/46419244-52de2980-c6e2-11e8-822b-1931bcf2d983.jpg)

![https everywhere ux 8](https://user-images.githubusercontent.com/28545431/46419245-52de2980-c6e2-11e8-9072-663084e3179e.jpg)

![https everywhere ux 9](https://user-images.githubusercontent.com/28545431/46419246-5376c000-c6e2-11e8-8e0e-c1e63cf0af69.jpg)

![https everywhere ux 10](https://user-images.githubusercontent.com/28545431/46419247-5376c000-c6e2-11e8-91e7-134b060c4bc9.jpg)

![https everywhere ux 11](https://user-images.githubusercontent.com/28545431/46419249-5376c000-c6e2-11e8-9776-ed30f8ba8bdc.jpg)

![https everywhere ux 12](https://user-images.githubusercontent.com/28545431/46419250-5376c000-c6e2-11e8-8f5d-0e50687392a1.jpg)

![https everywhere ux 13](https://user-images.githubusercontent.com/28545431/46419251-540f5680-c6e2-11e8-961f-431177e5f7b8.jpg)

![https everywhere ux 14](https://user-images.githubusercontent.com/28545431/46419252-540f5680-c6e2-11e8-84ce-e9cf24cfb2b4.jpg)

![https everywhere ux 15](https://user-images.githubusercontent.com/28545431/46419253-540f5680-c6e2-11e8-9e50-d62a8d0241f9.jpg)

![https everywhere ux 16](https://user-images.githubusercontent.com/28545431/46419254-540f5680-c6e2-11e8-804c-79f435642dd7.jpg)

![https everywhere ux 17](https://user-images.githubusercontent.com/28545431/46419255-54a7ed00-c6e2-11e8-9192-e187438b0f69.jpg)

![https everywhere ux 18](https://user-images.githubusercontent.com/28545431/46419256-54a7ed00-c6e2-11e8-8c7f-9009a7304e6d.jpg)

[//]: # (If you are reporting a ruleset/website problem, include the top-level)
[//]: # (domain below. For example, if you want to report an issue about)
[//]: # ("one.example.com" and "two.example.com", then the line below should)
[//]: # (be:)
[//]: #
[//]: # (Domain: example.com)
[//]: #
[//]: # (Be sure to remove the parenthesis and comment marker.  If you are only)
[//]: # (reporting an issue about "one.example.com", then the line below should)
[//]: # (be:)
[//]: #
[//]: # (Domain: one.example.com) 
[//]: #
[//]: # (Only include one top-level domain. If you have more than one top-level)
[//]: # (domain to report, such as both "example.com" and "example.org", open a)
[//]: # (new issue for each top-level domain.)
[//]: #
[//]: # (If you are not reporting a ruleset/website problem, you can delete the)
[//]: # ("Domain" line or leave it blank.)

[//]: # (Include any other relevant information below. Thank you again for)
[//]: # (helping to improve HTTPS Everywhere.)

**Comment by Hainish** [comment]
@glenn-sorrentino Thanks *so much* for putting this together.  I will have time to review it hopefully either tomorrow or the next day.  This is truly awesome work.

**Comment by glenn-sorrentino** [comment]
@Hainish My pleasure. I just updated the document with @cschanaj's suggestions.

@cschanaj - I added an accordion section for the URLs so that the use case you suggested is covered. I also added a section-by-section "Reset" so that it gives more flexibility to the user. Also good suggestion about how deep the other settings are - I added a "Settings" link in the last section of the menu! 

![updated-menu](https://user-images.githubusercontent.com/28545431/46419531-f7606b80-c6e2-11e8-8c51-30c9540b7387.png)

**Comment by zoracon** [comment]
@glenn-sorrentino  Just echoing @Hainish here but this is a great contribution! I was wondering what were your thoughts on the new "Disable HTTPS Everywhere on this site" feature. Do you picture a toggle would work for that as well?

**Comment by glenn-sorrentino** [comment]
@zoracon I have a toggle for that functionality, but the label is "Enable for this site" so that the user is turning off something that's on - I think it might be confusing to say "disable for this site" to turn on the off mode... make sense?

**Comment by zoracon** [comment]
> @zoracon I have a toggle for that functionality, but the label is "Enable for this site" so that the user is turning off something that's on - I think it might be confusing to say "disable for this site" to turn on the off mode... make sense?
Ah I missed that in the doc. Makes sense to me! Thank you!

**Comment by BenBE** [comment]
I'd prefer the options (Block unencrypted traffic) to also use a toggle for more consistency.

Also what about aligning the labels next to the toggles and the checkboxes, avoiding the step currently introduced by those elements being of different width.

**Comment by ghostwords** [comment]
Hi @glenn-sorrentino! We could really use your help over at HTTPS Everywhere's sibling project [Privacy Badger](https://github.com/EFForg/privacybadger). I think our biggest UX issue is that we still prominently show sliders in the popup even though we don't want most people to fiddle with them. This leads to general confusion about how Privacy Badger works, and, more specifically, [users unintentionally breaking the Web for themselves](https://github.com/EFForg/privacybadger/issues/2021). Thanks in advance for any help, totally understand if you are busy/not interested, and sorry for hijacking the thread!

**Comment by Bisaloo** [comment]
What's the status here? If everyone agrees, we should probably give @glenn-sorrentino the green light to submit a PR.

Unless the EFF own UX team is preparing some feedback on this?

**Comment by Hainish** [comment]
@Bisaloo thanks for keeping this thread alive.  Our UX team plans on sitting down with @glenn-sorrentino and talking over some of these changes.  Once this is done I plan on implementing it!

**Comment by zoracon** [comment]
As an update: Development for the menu will start soon (next week) to implement wire frames derived from this audit and internal UX research!

**Comment by KyleSanderson** [comment]
Cross-posting in here. In the future please run a campaign before making these insane regressions, you've hurt so many users and what shipped is UX Hell.

**Comment by andresbase** [comment]
I have removed some inappropriate comments.

**Comment by Hainish** [comment]
We've implemented UX changes based on this and other audits.  Thanks so much @glenn-sorrentino !
